### PR TITLE
makes inclusion of generated docs optional

### LINKED ISSUE #3818: Cannot  build master, Docs, asciidoctor:, datasource-guide.adoc: line 228: include file not found
Build fails on Documentation build: 
```
./mvnw install

[INFO] --- asciidoctor-maven-plugin:2.0.0-RC.1:process-asciidoc (output-html) @ quarkus-documentation ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 17 resources
[INFO] Rendered /home/karm/workspaceRH/quarkus/docs/src/main/asciidoc/getting-started-knative-guide.adoc
[INFO] asciidoctor: ERROR: datasource-guide.adoc: line 228: include file not found: /home/karm/workspaceRH/quarkus/docs/src/main/asciidoc/generated/quarkus-agroal.adoc
[INFO] Rendered /home/karm/workspaceRH/quarkus/docs/src/main/asciidoc/datasource-guide.adoc
[ERROR] asciidoctor: ERROR: datasource-guide.adoc: line 228: include file not found: /home/karm/workspaceRH/quarkus/docs/src/main/asciidoc/generated/quarkus-agroal.adoc
[INFO] ------------------------------------------------------------------------

```
**Environment:**
Linux, Java 11, Quarkus git HEAD: 7404e8e

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #3818

/cc @gsmet

### CODE REVIEW

File: docs/pom.xml [hunk 1]
```diff
@@ -38,7 +38,7 @@
                     <enableVerbose>true</enableVerbose>
                     <logHandler>
                         <failIf>
-                          <severity>INFO</severity>
+                          <severity>ERROR</severity>
```
Comment (gsmet) [inline docs/pom.xml]: Using `WARN` would be better.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/datasource-guide.adoc [hunk 2]
```diff
@@ -225,5 +225,5 @@ AgroalDataSource dataSource2;
 
 == Agroal Configuration Reference
 
-include::generated/quarkus-agroal.adoc[]
+include::generated/quarkus-agroal.adoc[opts=optional]
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource-guide.adoc]: Another remark, I don't think the TOC level is OK, is it? Can you check that?

We should add one toc level with `leveloffset=+1`.

**Review by gsmet — APPROVED** [review]

File: docs/src/main/asciidoc/hibernate-search-guide.adoc [hunk 3]
```diff
@@ -545,7 +545,7 @@ This is for testing purposes with the Elasticsearch Docker container.
 It should not be used in production.
 
 == Hibernate Search Configuration
-include::generated/quarkus-hibernate-search-elasticsearch.adoc[]
+include::generated/quarkus-hibernate-search-elasticsearch.adoc[leveloffset=+1, opts=optional]
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-search-guide.adoc]: Oh, I missed you changed the Search guide. I think we should consistently include the configuration reference at the end of the guide with a `Configuration Reference` title.

Considering how long it is, it kinda breaks the guide.

Here, we could just add a note saying `For more information about the Hibernate Search extension configuration, please refer to the link-to-anchor[Configuration Reference]` with a link to an anchor.

File: docs/src/main/asciidoc/datasource-guide.adoc [hunk 4]
```diff
@@ -223,7 +223,7 @@ AgroalDataSource dataSource1;
 AgroalDataSource dataSource2;
 --
 
-== Agroal Configuration Reference
+=== Agroal Configuration Reference
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource-guide.adoc]: Why? I think it was correct before.

File: docs/src/main/asciidoc/sending-emails.adoc [hunk 5]
```diff
@@ -63,7 +63,7 @@ quarkus.mailer.password=....
 
 The following table lists the properties that can be configured:
 
-include::generated/quarkus-mailer-impl.adoc[]
+include::generated/quarkus-mailer-impl.adoc[opts=optional, leveloffset=+1]
```
Comment (gsmet) [inline docs/src/main/asciidoc/sending-emails.adoc]: Same here, I would move the configuration reference at the end and add a note pointing to it.

File: docs/src/main/asciidoc/datasource-guide.adoc (hunk 4 above)
Comment (machi1990) [reply docs/src/main/asciidoc/datasource-guide.adoc]: Good catch. This should not have slipped in.

File: docs/src/main/asciidoc/hibernate-search-guide.adoc (hunk 3 above)
Comment (machi1990) [reply docs/src/main/asciidoc/hibernate-search-guide.adoc]: +1

Comment (machi1990) [reply docs/src/main/asciidoc/hibernate-search-guide.adoc]: A really good suggestion. Thanks. New changes pushed, WDYT?

File: docs/src/main/asciidoc/datasource-guide.adoc [hunk 6]
```diff
@@ -109,6 +109,9 @@ quarkus.datasource.max-size=13
 
 There are other configuration options, detailed below.
 
+[TIP]
+For more information about the Agroal extension configuration please refer to the <<configuration-reference, Configuration Reference>>
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource-guide.adoc]: Minor nitpick: we need a dot at the end of the sentence.

File: docs/src/main/asciidoc/datasource-guide.adoc [hunk 7]
```diff
@@ -223,7 +226,8 @@ AgroalDataSource dataSource1;
 AgroalDataSource dataSource2;
 --
 
-== Agroal Configuration Reference
 
-include::generated/quarkus-agroal.adoc[]
+[[configuration-reference]]
+== Agroal Configuration Reference
```
Comment (gsmet) [inline docs/src/main/asciidoc/datasource-guide.adoc]: I would add a new line before the include.

File: docs/src/main/asciidoc/hibernate-search-guide.adoc [hunk 8]
```diff
@@ -544,8 +544,8 @@ While, on a production setup, the `committed` default might be a suitable value,
 This is for testing purposes with the Elasticsearch Docker container.
 It should not be used in production.
 
-== Hibernate Search Configuration
-include::generated/quarkus-hibernate-search-elasticsearch.adoc[]
+[TIP]
+For more information about the Hibernate Search extension configuration please refer to the <<configuration-reference, Configuration Reference>>
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-search-guide.adoc]: Same remarks for all the guides :).

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a couple of additional comments

**Review by gsmet — APPROVED** [review]

**Pull request merged**